subsys/bluetooth/controller/Kconfig

```
@@ -635,6 +635,39 @@ config BT_CTLR_SDC_CS_EVENT_LEN_DEFAULT
 	help
 	  The time set aside for channel sounding events in microseconds.
 
+choice BT_CTLR_SDC_CS_T_PM_LEN_DEFAULT_SELECTION
+prompt "Default channel sounding t_pm length selection"
+	depends on BT_CTLR_CHANNEL_SOUNDING
+	default BT_CTLR_SDC_CS_T_PM_LEN_10_US
+	help
+	  Helper to select the time set aside for channel sounding tones in microseconds.
+
+config BT_CTLR_SDC_CS_T_PM_LEN_10_US
+	bool "Set default channel sounding t_pm length to 10 us"
+	depends on BT_CTLR_CHANNEL_SOUNDING
+	help
+	  Set controller default channel sounding tones to 10 microseconds.
+
+config BT_CTLR_SDC_CS_T_PM_LEN_20_US
+	bool "Set default channel sounding t_pm length to 20 us"
+	depends on BT_CTLR_CHANNEL_SOUNDING
+	help
+	  Set controller default channel sounding tones to 20 microseconds.
+
+config BT_CTLR_SDC_CS_T_PM_LEN_40_US
+	bool "Set default channel sounding t_pm length to 40 us"
+	depends on BT_CTLR_CHANNEL_SOUNDING
+	help
+	  Set controller default channel sounding tones to 40 microseconds.
+endchoice
+
+config BT_CTLR_SDC_CS_T_PM_LEN_DEFAULT
+	int "Default Channel Sounding t_pm length in microseconds"
```

why this, when you already have the choice ?

Also, this setting becomes stuck, meaning if someone open `menuconfig` and selects another value in the choice, then this setting will not be updated because it has a prompt.

https://docs.zephyrproject.org/latest/build/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig

---

KConfig does not have a system where you can have a choice parameter that sets a value, so instead we are setting a bool and using that bool to set a value. This is, to my understanding, the recommended way to have a choice between distinct integer values.

For the second issue would it be enough to change it to? 
```
config BT_CTLR_SDC_CS_T_PM_LEN_DEFAULT
	int "Default Channel Sounding t_pm length in microseconds" if !BT_CTLR_SDC_CS_T_PM_LEN_DEFAULT_SELECTION
	depends on BT_CTLR_CHANNEL_SOUNDING
	default 10 if BT_CTLR_SDC_CS_T_PM_LEN_10_US
	default 20 if BT_CTLR_SDC_CS_T_PM_LEN_20_US
	default 40 if BT_CTLR_SDC_CS_T_PM_LEN_40_US
```

---

> KConfig does not have a system where you can have a choice parameter that sets a value, so instead we are setting a bool and using that bool to set a value. 

Correct, my bad to not noticing the second was an integer.

In that case then just remove the prompt, that will avoid the stuck symbol syndrome.
```suggestion
	int
```